File: app/src/main/java/com/example/myapplication/DaftarActivity.java
package com.example.myapplication;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.content.Intent;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

public class DaftarActivity extends AppCompatActivity {

    EditText username, password, confpassword;
    Button Btndaftar;
    TextView txtLogin;
    ImageView Btnback;
    DBHelper DB;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_daftar);

        if (getSupportActionBar() != null) {
            getSupportActionBar().hide();
        }

        username = findViewById(R.id.username);
        password = findViewById(R.id.password);
        Btndaftar = findViewById(R.id.btn_daftar);
        txtLogin = findViewById(R.id.txt_login);
        confpassword = findViewById(R.id.conf_password);
        Btnback = findViewById(R.id.btn_back);
        DB = new DBHelper(this);

        Btndaftar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // Isi action button

                String Username = username.getText().toString();
                String Password = password.getText().toString();
                String Confpassword = confpassword.getText().toString();

                if (Username.equals("") || Password.equals("") || Confpassword.equals("")){
                    Toast.makeText(DaftarActivity.this, "Mohon isi semua kolom", Toast.LENGTH_SHORT).show();
                }else{
                    if (Password.equals(Confpassword)){
                        boolean checkuser = DB.checkusername(Username);
                        if (!checkuser){
                            boolean insert = DB.insertData(Username, Password);
                            if (insert){
                                Toast.makeText(DaftarActivity.this, "Pendaftaran berhasil!", Toast.LENGTH_SHORT).show();
                                Intent intent = new Intent(getApplicationContext(), MainActivity.class);
                                startActivity(intent);
                            }else{
                                Toast.makeText(DaftarActivity.this, "Pendaftaran gagal! Coba lagi", Toast.LENGTH_SHORT).show();
                            }
                        }else {
                            Toast.makeText(DaftarActivity.this, "Username sudah didaftarkan, segera login", Toast.LENGTH_SHORT).show();
                        }
                    }else{
                        Toast.makeText(DaftarActivity.this, "Password tidak sama", Toast.LENGTH_SHORT).show();
                    }
                }

                if (Username.isEmpty()) {
                    username.setError("Username/email wajib diisi");
                    return;
                }
                if (Password.isEmpty()) {
                    password.setError("Password harus diisi");
                    return;
                }
                if (Confpassword.isEmpty()) {
                    confpassword.setError("Konfirmasi password harus diisi");
                    return;
                }
            }
        });
        Btnback.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(DaftarActivity.this, MainActivity.class);
                startActivity(intent);
            }
        });

        txtLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // Ketika teks "Daftar" diklik, pindah ke DaftarActivity
                Intent intent = new Intent(DaftarActivity.this, MainActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/com/example/myapplication/DaftarActivity.kt
package com.example.myapplication

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.MotionEvent
import android.view.View
import android.widget.EditText
import android.text.method.HideReturnsTransformationMethod
import android.text.method.PasswordTransformationMethod
import kotlinx.android.synthetic.main.activity_daftar.*

class DaftarActivity : AppCompatActivity() {
    private var passwordVisible = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_daftar)
        supportActionBar?.hide()

        btnBackDaftarListener()
        txtLoginListener()
        passwordListener()
    }

    private fun btnBackDaftarListener() {
        btn_back.setOnClickListener {
            startActivity(Intent(this, MainActivity::class.java))
        }
    }

    private fun txtLoginListener() {
        txt_login.setOnClickListener {
            startActivity(Intent(this, MainActivity::class.java))
        }
    }

    private fun passwordListener() {
        val password = findViewById<EditText>(R.id.pass)

        password.setOnTouchListener { view, motionEvent ->
            val right = 2
            if (motionEvent.action == MotionEvent.ACTION_UP) {
                if (motionEvent.rawX >= password.right - password.compoundDrawables[right].bounds.width()) {
                    val selection = password.selectionEnd
                    if (passwordVisible) {
                        // Set drawable image here
                        password.setCompoundDrawablesRelativeWithIntrinsicBounds(0, 0, R.drawable.ic_baseline_visibility_off_24, 0)
                        // For hide password
                        password.transformationMethod = PasswordTransformationMethod.getInstance()
                        passwordVisible = false
                    } else {
                        // Set drawable image here
                        password.setCompoundDrawablesRelativeWithIntrinsicBounds(0, 0, R.drawable.ic_baseline_visibility_24, 0)
                        // For show password
                        password.transformationMethod = HideReturnsTransformationMethod.getInstance()
                        passwordVisible = true
                    }
                    password.setSelection(selection)
                    return@setOnTouchListener true
                }
            }
            false
        }
    }
}
